code/obj/flock/structure/flock_structure_parent.dm

```diff
@@ -49,6 +68,23 @@
 /obj/flock_structure/proc/process()
 	// override
 
+/obj/flock_structure/proc/groupcheck() //rechecks if the tile under's group matches its own
+	if(!usesgroups) return
+	if(istype(get_turf(src), /turf/simulated/floor/feather))
+		var/turf/simulated/floor/feather/f = get_turf(src)
+		if(src.grouptile == f && grouptile.group == src.group) return//no changes its all good
+		else if(!src.grouptile == f && f.group == src.group)//if the grouptile is different but the groups the same
+			src.grouptile = f//just move the connected tile, this should really rarely happen if the structure is moved somehow
+		else if(!src.grouptile == f && !f.group == src.group)//if both stuff is different.
+			src.grouptile = f
+			src.group?.removestructure(src)
+			src.group = f.group
+			src.group?.addstructure(src)
+		else if(src.grouptile == f && !grouptile.group == src.group)//if just the tile's group is different
+			src.group?.removestructure(src)
+			src.group = grouptile.group
+			src.group?.addstructure(src)
```

Comment:
why do you keep ?. checking group, even after literally setting it

Reply:
Oops, force of habit